### ISSUE TITLE
Problem with Docker compose

### ISSUE THREAD

**Issue by giampaolotrapasso** [confluentinc/bottledwater-pg#78]
Hi to all,
I'm following the Readme to install all the stuff but I'm quite new to Docker. I have a warning on the postgres install and then on connecting related to the port of the database. In the specific I have this:

```
giampaolo@rbbox:~/dev/bottledwater-pg$ docker-compose up -d zookeeper kafka schema-registry postgres
WARNING: The POSTGRES_PORT_5432_TCP_ADDR variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PORT_5432_TCP_PORT variable is not set. Defaulting to a blank string.
bottledwaterpg_zookeeper_1 is up-to-date
bottledwaterpg_postgres_1 is up-to-date
bottledwaterpg_kafka_1 is up-to-date
bottledwaterpg_schema-registry_1 is up-to-date
giampaolo@rbbox:~/dev/bottledwater-pg$ docker-compose run --rm postgres psqlWARNING: The POSTGRES_PORT_5432_TCP_ADDR variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PORT_5432_TCP_PORT variable is not set. Defaulting to a blank string.
psql: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```

How/when do I configure those variables? There's no mention in the Readme. 
Before doing compose up, I did
```
export POSTGRES_PORT_5432_TCP_PORT=5432
export  POSTGRES_PORT_5432_TCP_ADDR=172.17.0.1
````

but this command
` docker inspect -f '{{.Name}} - {{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $(docker ps -aq)`
returned me
````
/bottledwaterpg_schema-registry_1 - 172.17.0.5
/bottledwaterpg_kafka_1 - 172.17.0.4
/bottledwaterpg_zookeeper_1 - 172.17.0.3
/bottledwaterpg_postgres_1 - 172.17.0.2
````
So I do not understand if during compose-up the env variables are used or not, and if I have to change them after compose-up. With proper indications, I would be glad to submit a PR with more details in the Readme file for future newcomers.
 
Thanks in advance.

**Comment by indyaah** [comment]
@giampaolotrapasso : you are running wrong docker command to connect to your database, in fact there is no such container named postgres running.

Try `docker exec -it bottledwaterpg_postgres_1 bash` and you would be in your postgres contain
This basically is sort of ssh'ing to running postgres, you can read more here https://docs.docker.com/engine/reference/commandline/exec/.

**Comment by giampaolotrapasso** [comment]
@indyaah: no problem. Finally I got all the stuff working. Next week I'll document it into a gist the exact steps I did. I'm sure I did not all things right but probably you can address me to my blind spots.

**Comment by alwaysanirudh** [comment]
Hi @giampaolotrapasso I'm facing the same issue. Can you guide me with this?

**Comment by giampaolotrapasso** [comment]
_disclaimer 1: I'm a docker newbie. This is a dirty workaround I did._
_disclaimer 2: I'm just pasting things from history to give you a quick reply YMMV, meanwhile I'll do a PR or a gist._

This is my `docker-compose.yaml`, double check diffs:
````
# N.B. assumes you have set the environment variable
# KAFKA_ADVERTISED_HOST_NAME, in the environment of the host where you are
# running docker-compose, to the IP address of the Docker host.
#
# You can determine the Docker host IP with a command like:
#
#   docker run --rm debian:latest ip route | awk '/^default via / { print $3 }'

zookeeper:
  image: confluent/zookeeper
  hostname: zookeeper
  ports:
    - '2181:2181'
kafka:
  image: confluent/kafka:0.9.0.0-cp1
  hostname: kafka
  links:
    - zookeeper
  environment:
    KAFKA_LOG_CLEANUP_POLICY:
    KAFKA_ADVERTISED_HOST_NAME:
    KAFKA_AUTO_CREATE_TOPICS_ENABLE:
  ports:
    - '9092:9092'
schema-registry:
  image: confluent/schema-registry
  hostname: schema-registry
  links:
    - zookeeper
    - kafka
  ports:
    - '48081:8081'
  environment:
    SCHEMA_REGISTRY_AVRO_COMPATIBILITY_LEVEL: none
postgres:
  build: ./tmp
  dockerfile: Dockerfile.postgres
  hostname: postgres
  ports:
    - '45432:5432'
  environment:
    POSTGRES_PORT_5432_TCP_PORT: 5432
bottledwater-json:
  build: ./tmp
  dockerfile: Dockerfile.client
  hostname: bottledwater
  links:
    - postgres
    - kafka
  command: --output-format=json --allow-unkeyed
bottledwater-avro:
  build: ./tmp
  dockerfile: Dockerfile.client
  hostname: bottledwater
  links:
    - postgres
    - kafka
    - schema-registry
  command: --output-format=avro --allow-unkeyed
psql:
  image: postgres:9.5
  links:
    - postgres
  command: 'sh -c ''exec psql -h "$POSTGRES_PORT_5432_TCP_ADDR" -p "$POSTGRES_PORT_5432_TCP_PORT" -U postgres'''
topiclist:
  image: confluent/tools
  links:
    - zookeeper
    - kafka
  entrypoint: ['/usr/bin/kafka-topics']
kafka-avro-console-consumer:
  image: confluent/tools
  links:
    - zookeeper
    - kafka
    - schema-registry
  entrypoint: ['/usr/bin/kafka-avro-console-consumer']
kafka-tools:
  image: confluent/tools
  links:
    - zookeeper
    - kafka
  entrypoint: /usr/bin/confluent-tools
```

add to your `.profile` and refresh shell enviroment
```bash
export POSTGRES_PORT_5432_TCP_PORT=5432
export POSTGRES_PORT_5432_TCP_ADDR=172.17.0.2
export SCHEMA_REGISTRY_PORT_8081_TCP_ADDR=172.17.0.5
export SCHEMA_REGISTRY_PORT_8081_TCP_PORT=8081
export KAFKA_PORT_9092_TCP_ADDR=172.17.0.4
export KAFKA_PORT_9092_TCP_PORT=9092 
```
_This is the part where I need to study docker-compose, I do not want to cable IPs._


Run
`docker-compose up -d  postgres``

on another terminal run
`docker-compose run --rm  psql``
and install extensions as readme.

run
`docker-compose up -d zookeeper`

then run
`docker-compose up  kafka-tools`

then run
`docker-compose up -d schema-registry`

then run
`docker-compose up -d kafka`

Now you should run 
````
radicalbit@rbbox:~/dev/bottledwater-pg$ docker inspect -f '{{.Name}} - {{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $(docker ps -aq)
```

and having something like that, pay attention to IPs, they should be coherent with the `.profile`.
```
/bottledwaterpg_kafka-tools_1 - 
/bottledwaterpg_schema-registry_1 - 172.17.0.5
/bottledwaterpg_kafka_1 - 172.17.0.4
/bottledwaterpg_zookeeper_1 - 172.17.0.3
/bottledwaterpg_postgres_1 - 172.17.0.2
````

Now you can run

` docker-compose up -d bottledwater-avro`

and on a different terminal

`docker-compose run --rm kafka-avro-console-consumer --from-beginning --topic test --zookeeper 172.17.0.3:2181 --property schema.registry.url=http://172.17.0.5:8081`
_putting env variable in the compose file would simplify line command parametters_

Do CUD operations on the test table and you should see action on the consumer terminal.

**Comment by giampaolotrapasso** [comment]
@alwaysanirudh: ah addition. You can run almost everything with a single compose up. I'm reading about version 2 of compose file format. I will do a gist with that version.

**Comment by samstokes** [comment]
Hi @giampaolotrapasso @alwaysanirudh, sorry that you've been having trouble with this, and for a very belated reply.  In theory you shouldn't need to set those environment variables manually (`POSTGRES_PORT_5432_TCP_PORT` etc), Docker should automatically configure them.

I think you may be using a newer version of Docker and/or Docker Compose than we have previously tested with.  Docker made some changes to inter-container networking and we may need to update the Docker setup to work properly for newer versions.  Could you paste the output of `docker version` and `docker-compose version`?  Here are mine for comparison:

```bash
$ docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64

$ docker-compose version
docker-compose version: 1.4.2
docker-py version: 1.3.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```

**Comment by giampaolotrapasso** [comment]
Hi @samstokes, these are my settings.
````
[Giampaolo - 2016-06-08 21:11:13] ~/> docker version
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:44:17 2016
 OS/Arch:      darwin/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
[Giampaolo - 2016-06-08 21:11:18] ~/> docker-compose version
docker-compose version 1.7.1, build 0a9ab35
docker-py version: 1.8.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1j 15 Oct 2014
[Giampaolo - 2016-06-08 21:11:24] ~/> 
````

I've managed to make work bottledwater-pg with a slightly modified version of `docker-compose.yaml`file that you can find in my fork.

**Comment by samstokes** [comment]
Hi @giampaolotrapasso @alwaysanirudh, I merged some changes to master (#88) that should make the build work better with newer versions of Docker and Docker Compose.  I think it should resolve the problems you were seeing.  Do you still see these issues after updating to the latest master?
